gobblin-core/src/main/java/org/apache/gobblin/writer/partitioner/TimeBasedORCSerDeWriterPartitioner.java

```diff
@@ -0,0 +1,165 @@
+ *
+ *
+ * If a record contains none of the specified fields, or if no field is specified, the current timestamp will be used.
+ */
+public class TimeBasedORCSerDeWriterPartitioner extends TimeBasedWriterPartitioner<Object> {
+
+  private static final String WRITER_PARTITION_COLUMN_NAME = ConfigurationKeys.WRITER_PREFIX + ".partition.column";
+  private static final String WRITER_PARTITION_COLUMN_INDEX =
+      ConfigurationKeys.WRITER_PREFIX + ".partition.column.index";
+  private static final String WRITER_PARTITION_COLUMN_PATTERN =
+      ConfigurationKeys.WRITER_PREFIX + ".partition.column.pattern";
+  private static final String WRITER_PARTITION_COLUMN_TIMEZONE =
+      ConfigurationKeys.WRITER_PREFIX + ".partition.column.timezone";
+
+  private static final String UTC = "UTC";
+  private static final String ORC_SERDE_INSPECTOR = "getInspector";
+  private static final String ORC_SERDE_ROW = "realRow";
+
+  private final Optional<String> partitionColumnName;
+  private final Optional<Integer> partitionColumnIndex;
+  private final Optional<String> partitionColumnPattern;
+  private final Optional<String> partitionColumnTimeZone;
+  private DateTimeFormatter partitionColumnDateTimeFormatter;
+
+  public TimeBasedORCSerDeWriterPartitioner(State state, int numBranches, int branchId) {
+    super(state, numBranches, branchId);
+    this.partitionColumnName = getPropertyAsString(WRITER_PARTITION_COLUMN_NAME, state, numBranches, branchId);
+    this.partitionColumnIndex = getPropertyAsInteger(WRITER_PARTITION_COLUMN_INDEX, state, numBranches, branchId);
+    this.partitionColumnPattern = getPropertyAsString(WRITER_PARTITION_COLUMN_PATTERN, state, numBranches, branchId);
+    this.partitionColumnTimeZone = getPropertyAsString(WRITER_PARTITION_COLUMN_TIMEZONE, state, numBranches, branchId);
+    if (this.partitionColumnPattern.isPresent()) {
+      this.partitionColumnDateTimeFormatter = DateTimeFormatter.ofPattern(partitionColumnPattern.get());
+    }
+  }
+
+  private static Optional<Integer> getPropertyAsInteger(String property, State state, int numBranches, int branchId) {
+    String propName = ForkOperatorUtils.getPropertyNameForBranch(property, numBranches, branchId);
+    return state.contains(propName) ? Optional.of(state.getPropAsInt(propName)) : Optional.empty();
+  }
+
+  private static Optional<String> getPropertyAsString(String property, State state, int numBranches, int branchId) {
+    String propName = ForkOperatorUtils.getPropertyNameForBranch(property, numBranches, branchId);
+    return state.contains(propName) ? Optional.of(state.getProp(propName)) : Optional.empty();
+  }
+
+  /**
+   * Check if the partition column value is present and is a Long object.
+   * if partitionColumnPattern is present we format LocalDateTime as EpochMilli.
+   * Otherwise, use current system time.
+   */
+  private long getRecordTimestamp(Optional<Object> writerPartitionColumnValue) {
```

Comment:
It seems that this method is almost the same as that in Avro partitioner `getRecordTimestamp` method. Please try to reduce the redundancy by abstracting it out.